src/main/python/camdkit/framework.py

```diff
@@ -233,30 +312,338 @@ def make_json_schema() -> dict:
       "additionalProperties": False
     }
 
-class StrictlyPositiveIntegerParameter(Parameter):
+class IntegerParameter(Parameter):
+  """Base class for integer parameters"""
+
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return value
+
+  @staticmethod
+  def from_json(value: typing.Any) -> typing.Any:
+    return int(value)
+  
+class NonNegativeIntegerParameter(IntegerParameter):
 
   @staticmethod
   def validate(value) -> bool:
     """The parameter shall be a integer in the range (0..2,147,483,647]."""
 
+    return isinstance(value, numbers.Integral) and value >= 0
+
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": INT_MAX
+    }
+  
+class StrictlyPositiveIntegerParameter(IntegerParameter):
+
+  @staticmethod
+  def validate(value) -> bool:
+    """The parameter shall be a integer in the range (1..2,147,483,647]."""
+
     return isinstance(value, numbers.Integral) and value > 0
 
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "integer",
+      "minimum": 1,
+      "maximum": INT_MAX
+    }
+  
+class NonNegativeRealParameter(Parameter):
+  
+  @staticmethod
+  def validate(value) -> bool:
+    """The parameter shall be a non-negative real number."""
+
+    return isinstance(value, numbers.Real) and value >= 0.0
+
   @staticmethod
   def to_json(value: typing.Any) -> typing.Any:
     return value
 
   @staticmethod
   def from_json(value: typing.Any) -> typing.Any:
-    return int(value)
+    return float(value)
+  
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "number",
+      "minimum": 0.0,
+    }
+  
+class EnumParameter(StringParameter):
```

Comment:
Shouldn't this be derived from Python enums? In any case, why is `enum_class` necessary since the name of the derived class can be accessed?

Reply:
It is represented as a string in json schema with an enum [...] property. (The enum_class does derive from the Python Enum.) I will determine the enum_class from the class name as you suggest.